src/components/date/date.spec.tsx

```diff
@@ -558,10 +667,10 @@ describe("Date", () => {
   describe("initial value as ISO format", () => {
     describe.each(
       Object.keys(locales).filter((l) => !["en-CA", "en-US"].includes(l))
```

Comment:
```suggestion
      Object.keys(locales).filter((l) => !["en-CA", "en-US"].includes(l))
```
```suggestion
      (Object.keys(locales) as (keyof typeof locales)[])filter((l) => !["en-CA", "en-US"].includes(l))
```

**suggestion** (non-blocking): I prefer the type assertion here instead of line 673 (where it should be removed from if you agree to this change) since here it's pretty obvious when reading it that it's a valid asertion, whereas it's less so on line 673.